packages/ui/src/components/commit-suggestions-dialog/commit-suggestions-dialog.tsx

```diff
@@ -0,0 +1,109 @@
+import { FC } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+
+import {
+  Button,
+  ButtonGroup,
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  Input,
+  Textarea
+} from '@/components'
+import { UsererrorError } from '@/types'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+
+export interface CommitSuggestionsFormType {
+  message?: string
+  title: string
+}
+
+const commitSuggestionsSchema = z.object({
+  message: z.string().optional(),
+  title: z.string()
+})
+
+export type CommitSuggestionsFormFields = z.infer<typeof commitSuggestionsSchema>
+
+export interface CommitSuggestionsDialogProps {
+  isOpen: boolean
+  commitTitlePlaceHolder?: string
+  // TODO: We need to decide how to display errors from the API.
+  error?: UsererrorError
+  onClose: () => void
+  onFormSubmit: (formValues: CommitSuggestionsFormType) => Promise<void>
+  isSubmitting: boolean
+}
+
+export const CommitSuggestionsDialog: FC<CommitSuggestionsDialogProps> = ({
+  isOpen,
+  onClose,
+  commitTitlePlaceHolder,
+  onFormSubmit,
+  isSubmitting
+}) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid }
+  } = useForm<CommitSuggestionsFormFields>({
+    resolver: zodResolver(commitSuggestionsSchema),
+    mode: 'onChange',
+    defaultValues: {
+      message: '',
+      title: commitTitlePlaceHolder
+    }
+  })
+
+  const isDisabledSubmission = isSubmitting || !isValid
+
+  const onSubmit: SubmitHandler<CommitSuggestionsFormFields> = data => {
+    if (isDisabledSubmission) return
+
+    onFormSubmit(data as CommitSuggestionsFormType)
+  }
+
+  return (
+    <Dialog open={isOpen} onOpenChange={onClose}>
+      <DialogContent className="max-w-[576px]">
+        <DialogHeader>
+          <DialogTitle>Commit Changes</DialogTitle>
+        </DialogHeader>
+
+        <form className="flex flex-col gap-y-7 pb-4" onSubmit={handleSubmit(onSubmit)}>
+          <Input
+            id="title"
+            label="Commit Message"
+            {...register('title')}
+            placeholder={commitTitlePlaceHolder ?? 'Add a commit message'}
```

Comment:
in a followup pr, we should probably go through all the pr pages and put strings into views.json

Reply:
agreed